tests/CMakeLists.txt

```diff
@@ -7,6 +7,8 @@ enable_testing()
 set(MODEL_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/)
 include_directories("${CMAKE_CURRENT_SOURCE_DIR}/../")
 
+find_package(Catch2 REQUIRED)
```

Comment:
Since Catch is part of the repo, you have to specify it's location:

https://github.com/catchorg/Catch2/blob/devel/docs/cmake-integration.md#cmake-target

see the second example.

Reply:
I will look into this early morning it's a bit late here and i am also on my phone so a bit hard to keep track.

Reply:
So whenever you have time.

Reply:
Since Catch2 is probably not installed on the system, you have to remove that line and add `Catch.cmake` and `Catch.cmake` to https://github.com/mlpack/models/tree/master/CMake

See https://github.com/catchorg/Catch2/blob/313071e8fee34bc96c98f0fa3c4065f39ed63236/docs/cmake-integration.md#automatic-test-registration for more information.